test-specifications/cryptoService/accountCreateTransaction.md

```diff
@@ -0,0 +1,155 @@
+# AccountCreateTransaction - Test specification
+
+## Description:
+This test specification for AccountCreateTransaction is to be one of many for testing the functionality of the Hedera SDKs. The SDK under test will use the language specific JSON-RPC server return responses back to the test driver.
+
+## Design:
+Each test within the test specification is linked to one of the properties within AccountCreateTransaction. Each property is tested with a mix of boundaries. The inputs for each test are a range of valid, minimum, maximum, negative and invalid values for the method. The expected response of a passed test can be a correct error response code or seen as the result of node queries. Success on the consensus node can be obtained by a queries such as AccountInfoQuery or AccountBalanceQuery, and on the mirror node through the rest API. Error codes are obtained from the response code proto files.
+
+**Transaction properties:**
+
+https://docs.hedera.com/hedera/sdks-and-apis/sdks/accounts-and-hbar/create-an-account
+
+**CryptoCreate protobufs:**
+
+https://github.com/hashgraph/hedera-protobufs/blob/main/services/crypto_create.proto
+
+**Response codes:**
+
+https://github.com/hashgraph/hedera-protobufs/blob/main/services/response_code.proto
+
+**Mirror Node APIs:**
+
+https://docs.hedera.com/hedera/sdks-and-apis/rest-api
+
+## Initialisation
+
+```jsx
+new AccountCreateTransaction()
+
+//example
+
+const transaction = new AccountCreateTransaction()
+  .setKey(privateKey.publicKey)
+  .setInitialBalance(new Hbar(1000))
+  ...
+```
+
+## JSON-RPC API Endpoint Documentation
+
+### Method Name
+
+`createAccount`
+
+### Parameters
+
+| Parameter Name            | Type   | Required/Optional | Description/Notes                                                                                                                           |
+|---------------------------|--------|-------------------|---------------------------------------------------------------------------------------------------------------------------------------------|
+| key                       | string | optional          | DER-encoded hex string representation for private or public keys. Keylists and threshold keys are the hex of the serialized protobuf bytes. |
```

Comment:
When we say required or optional do we mean this field is required to have the transaction succeed in normal conditions? The key field is required to be set when creating an account otherwise an account will not be created.

Reply:
This is saying that the `key` field is optional in the JSON RPC API. This was made optional to test the case if an `AccountCreateTransaction` were to be submitted without a set key. This doesn't speak to if the key is required by the consensus node or not, simply if the JSON RPC server should require this field every time it receives a `createAccount` method call. 

The idea with this table was for a developer developing their respective SDK's TCK server, they could take a look at this list to see the parameters for the method they need to develop. With `key` being optional, they would develop their method to then not expect to receive a `key` with every `createAccount` method call from the TCK test driver.

Reply:
Ah okay thanks for the explanation that makes sense!!